File: jna/src/main/java/dark/box.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;

public class box extends Structure  implements Structure.ByValue{
    public float x;
    public float y;
    public float w;
    public float h;

    public box() {
        super();
    }

    public box(Pointer peer) {
        super(peer);
        read();
    }

    public box(float x,float y,float w,float h){
        this.x = x;
        this.y = y;
        this.w = w;
        this.h = h;
    }

    public box(float[] bbox){
        x = bbox[0];
        y = bbox[1];
        w = bbox[2];
        h = bbox[3];
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("x","y","w","h");
    }

    public static class ByReference extends detection implements Structure.ByReference {
        public ByReference() { }
        public ByReference(Pointer p) { super(p); read(); }
    }

    public static class ByValue extends box implements Structure.ByValue { }
}

File: jna/src/main/java/dark_kotlin/box.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure


open class box : Structure, Structure.ByValue {
    @JvmField var x = 0f
    @JvmField var y = 0f
    @JvmField var w = 0f
    @JvmField var h = 0f

    constructor() : super()
    constructor(peer: Pointer) : super(peer) {
        read()
    }

    constructor(x: Float, y: Float, w: Float, h: Float) {
        this.x = x
        this.y = y
        this.w = w
        this.h = h
    }

    constructor(bbox: Array<Float>) {
        x = bbox[0]
        y = bbox[1]
        w = bbox[2]
        h = bbox[3]
    }

    override fun getFieldOrder(): List<String> {
        return listOf("x", "y", "w", "h")
    }

    class ByValue : box(), Structure.ByValue
    class ByReference() : detection(), Structure.ByReference {
        init {
            read()
        }
    }
}
